examples/Microsoft.Spark.CSharp.Examples/Batch/Logging.cs

```diff
@@ -0,0 +1,179 @@
+            trueSpam.CreateOrReplaceTempView("TrueSpamLogs");
+            DataFrame trueSpamSql = spark.Sql(
+                "SELECT truespamlogs.value FROM truespamlogs");
+
+            // Explore the columns in the data we have filtered
+            // Let's try getting the number of GET requests
+            IEnumerable<Row> rows = trueSpamSql.Collect();
+            int numGetRequests = 0;
+            foreach (Row row in rows)
+            {
+                string rowstring = row.ToString();
+                numGetRequests += ParseLog(rowstring);
+            }
+
+            Console.WriteLine("Number of GET requests: " + numGetRequests);
+
+            spark.Stop();
+        }
+
+        public static bool RegTest(string logLine, string regexType)
+        {
+            // Example Apache log line:   
+            // 64.242.88.10 - - [07/Mar/2004:16:47:12 -0800] "GET /robots.txt HTTP/1.1" 200 68
+            // 1:IP   2:client   3:user   4:date time   5:method   
+            // 6:req   7:proto   8:respcode   9:size
+            string useRx =
+                "^(\\S+) (\\S+) (\\S+) \\[([\\w:/]+\\s[+\\-]\\d{4})\\] \"(\\S+) (\\S+) (\\S+)\" (\\d{3}) (\\d+)";
+
+            // Which regex to use based on what we're filtering for
+            if (regexType == "ipfilter")
+            {
+                useRx = "^(?=10)";
+            }
+            else if (regexType == "spamfilter")
+            {
+                useRx = "\\b(?=spam)\\b";
+            }
+
+            return Regex.IsMatch(logLine, useRx);
+        }
+
+        public static int ParseLog(string logLine)
+        {
+            Match match = Regex.Match(
+                logLine,
+                "^(\\S+) (\\S+) (\\S+) \\[([\\w:/]+\\s[+\\-]\\d{4})\\] \"(\\S+) (\\S+) (\\S+)\" (\\d{3}) (\\d+)");
+
+            int groupCtr = 0;
+            int entryCtr = 0;
+            int numGets = 0;
+
+            string[] colNames =
+                {"ip",
+                "client",
+                "user",
+                "date",
+                "method",
+                "endpt",
+                "protocol",
+                "response",
+                "size"};
+
+            // Print out the full log line and then divide based on column
+            foreach (Group group in match.Groups)
+            {
+                if (groupCtr == 0)
+                {
+                    Console.WriteLine("Full log entry: '{0}'", group.Value);
+                    ++groupCtr;
+                    continue;
+                }
+                Console.WriteLine("    {0}: '{1}'", colNames[entryCtr], group.Value);
+
+                // Determine if it's a GET request
+                if (colNames[entryCtr] == "method")
+                {
+                    if (group.Value == "GET")
+                    {
+                        ++numGets;
+                    }
+                }
+                ++entryCtr;
+            }
+            return numGets;
+        }
```

Comment:
This seems a bit overly complicated for what it's doing.  Can we simplify this ?

Reply:
My challenge was finding a way to easily parse a completely text-based file (not csv or json) just based on pattern matching. I'll take a look at trying to simplify it!

Reply:
It seems to me that `public static int ParseLog(string logLine)` only reads one log line.  The number of `GET`s will either be 1 or 0.   Can't you simplify the entire method to be something similar to the following:

```csharp
public static int ParseLog(string logLine)
{
    Match match = Regex.Match(logLine, apacheRx);

    // add Console.WriteLine as needed.
    if (match.Success)
    {
        if (match.Groups[5] == "GET")
        {
            return 1;
        }
    }

    return 0;
}
```

Reply:
Yes you're right. I spent so much time looking at that original code that I needed the fresh perspective to help out here, so thank you :)